.github/workflows/_build_docker.yaml

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2022 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+# This reuseable workflow builds and tests the Macaron Docker image.
+
+name: Build and push Docker image
+on:
+  workflow_call:
+    inputs:
+      artifact-name:
+        required: true
+        type: string
+      artifact-sha256:
+        required: true
+        type: string
+permissions:
+  contents: read
+
+jobs:
+  build-docker:
+    runs-on: ubuntu-latest
+    permissions:
+      packages: read
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
+      with:
+        fetch-depth: 0
+
+    - name: Download artifact
+      uses: actions/download-artifact@9782bd6a9848b53b110e712e20e42d89988822b7 # v3.0.1
+      with:
+        name: ${{ inputs.artifact-name }}
+        path: dist
+
+    # Verify hashes by first computing hashes for the artifacts and then comparing them
+    # against the hashes for the artifact.
+    - name: Verify the artifact hash
+      env:
+        ARTIFACT_HASH: ${{ inputs.artifact-sha256 }}
+      run: |
+        set -euo pipefail
+        cd dist
```

Comment:
This line is needed because the artifact is downloaded into `dist/artifact_name/`. So the result of the download is
```
dist
└── artifact-ubuntu-latest-python-3.11
    ├── dist
    │   ├── macaron-0.0.0-build-epoch.txt
    │   ├── macaron-0.0.0-docs-html.zip
    │   ├── macaron-0.0.0-py3-none-any.whl
    │   ├── macaron-0.0.0-requirements.txt
    │   ├── macaron-0.0.0-sbom-go.json
    │   ├── macaron-0.0.0-sbom.json
    │   └── macaron-0.0.0.tar.gz
    └── src
        └── macaron
            └── bin
                ├── actionparser
                └── bashparser
```
Without changing the directory into dist, verifying the hash will fail as we don't reflect the exact directory structure that the hash is originally computed from. 
I tried following the existing [step](https://github.com/oracle-samples/macaron/blob/9e89477f6ad91749bbb9bda1d7361ee0d1cb3305/.github/workflows/release.yaml#L106) but it didn't work in this case.

Reply:
See my other comment. Didn't we decided to remove the Go binaries as they are included in the wheel? If we don't add those binaries, we shouldn't need to run `cd dist` I think.

Reply:
> Didn't we decided to remove the Go binaries as they are included in the wheel?
Yes we did. However, because how the artifact hash is calculated [here](https://github.com/oracle-samples/macaron/blob/9e89477f6ad91749bbb9bda1d7361ee0d1cb3305/.github/workflows/_build.yaml#L118) so I decided to keep it in the uploaded artifact. Perhaps it's better if we remove the steps to calculate the hash for both `actionparser` and `bashparser`.

Reply:
> Perhaps it's better if we remove the steps to calculate the hash for both `actionparser` and `bashparser`.

Yes, please remove them.

Reply:
Fixed in https://github.com/oracle-samples/macaron/pull/76/commits/d2fe70dc8bc12b52071d18e885bf63b86f9dcc25